### PR TITLE
Hydrate the contact bound to action reports

### PR DESCRIPTION
**Pull request by garethbowen**
# Description

This allows action reports to use the contact and parents of the
report subject.

medic/medic-webapp#3912

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Announced in Changes.md in plain English. Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

### CODE REVIEW

File: static/js/services/lineage-model-generator.js [hunk 1]
```diff
@@ -87,16 +87,18 @@ angular.module('inboxServices').factory('LineageModelGenerator',
       contact: function(id, options) {
         options = options || {};
         return get(id).then(function(docs) {
-          // the first row is the contact
-          var doc = docs.shift();
-           // everything else is the lineage
-          return hydrate(docs).then(function(lineage) {
-            var result = { _id: id };
+          return hydrate(docs).then(function(docs) {
```
Comment (estellecomment) [inline static/js/services/lineage-model-generator.js]: If i'm getting this right, this hydrating of everything including the "root" doc is new, and not tested AFAICT. (I'm relying heavily on tests for examples of what this whole file does, cos I'm not going to re-understand the code every time I have a doubt)

File: static/js/services/lineage-model-generator.js [hunk 2]
```diff
@@ -109,17 +111,17 @@ angular.module('inboxServices').factory('LineageModelGenerator',
        */
       report: function(id) {
         return get(id).then(function(docs) {
-          // the first row is the report
-          var doc = docs.shift();
-          // the second row is the report's contact
-          var contact = docs.shift();
-          // everything else is the lineage
-          return hydrate(docs).then(function(lineage) {
+          return hydrate(docs).then(function(docs) {
```
Comment (estellecomment) [inline static/js/services/lineage-model-generator.js]: same here

**Comment by estellecomment** [comment]
oh, another point : how much is the webapp hydration code now diverging from the sentinel hydration code? Can we make a note of how it's different? (and prioritize https://github.com/medic/medic-webapp/issues/3826, I see that it's not in any project, it's getting seriously horrible)

**Comment by garethbowen** [comment]
Added a test to cover this.

**Pull request merged**